core/src/test/kotlin/org/evomaster/core/search/gene/GeneUtilsEscapeTest.kt

```diff
@@ -10,28 +11,36 @@ internal class GeneUtilsEscapeTest {
 
     @Test
     fun testEscapes(){
-        val testJson = "{\n" +
+        /*val testJson = "{\n" +
                 "\"name\" : \"contains\\\"now\"\n" +
                 "\"values\" : \"various\u001dother\u0016unicode\"\n" +
                 "\"kotlinTest\" : \"some\$kotlin\"\n" +
             "}"
 
+         */
+        val testJson2 = """{"name":"kotlin"}"""
+
+        val testJson = StringGene("Quote\\Gene", "Test For the quotes${'$'}escape")
+
         val formatKotlin = OutputFormat.KOTLIN_JUNIT_5
         val formatJava5 = OutputFormat.JAVA_JUNIT_5
         println("TestJson Raw: $testJson")
-        println("TestJson Kotlin: ${GeneUtils.applyEscapes(testJson, "json", formatKotlin)}")
-        println("TestJson Java: ${GeneUtils.applyEscapes(testJson, "json", formatJava5)}")
+        println("TestJson Kotlin: ${testJson.getValueAsPrintableString(mode = "json", targetFormat = formatKotlin)}")
+        println("TestJson Java: ${testJson.getValueAsPrintableString(mode = "json", targetFormat = formatKotlin)}")
 
-        assertTrue(GeneUtils.applyEscapes(testJson, "json", formatKotlin).contains("\\"))
-        assertTrue(GeneUtils.applyEscapes(testJson, "json", formatJava5).contains("\\"))
+        //assertTrue(testJson.getValueAsPrintableString(mode = "json", targetFormat = formatKotlin).contains("\\"))
+        //assertTrue(testJson.getValueAsPrintableString(mode = "json", targetFormat = formatKotlin).contains("\\"))
 
-        assertTrue(GeneUtils.applyEscapes(testJson, "json", formatKotlin).contains("\""))
-        assertTrue(GeneUtils.applyEscapes(testJson, "json", formatKotlin).contains("\$"))
+        assertTrue(testJson.getValueAsPrintableString(mode = "json", targetFormat = formatKotlin).contains("\""))
```

Comment:
isn't `mode` an enum now instead of a string?

Reply:
I didn't refactor the Gene (and its methods) to take an EscapeMode instead of a string, so as not to damage other bits of the code. I can do that refactoring now.